### PR TITLE
Fix float32 support in denoise_bilateral and denoise_tv_bregman

### LINKED ISSUE #3449: Denoise throws ValueError
## Description

`_denoise_cy.pyx` throws a `ValueError` saying a `double` was expected but a `float` was received.

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/aether/programs/anaconda3/envs/cellprofiler/lib/python2.7/site-packages/skimage/restoration/_denoise.py", line 117, in denoise_bilateral
    bins, mode, cval)
  File "skimage/restoration/_denoise_cy.pyx", line 114, in skimage.restoration._denoise_cy._denoise_bilateral
ValueError: Buffer dtype mismatch, expected 'double' but got 'float'
```

## Way to reproduce

- [X] Code example
```
import numpy as np
import skimage.restoration

sigma = 16.0
sigma_range = .2
np.random.seed(0)
image = np.random.uniform(size=(100, 100)).astype(np.float32)
mask = np.ones(image.shape, bool)
mask[40:60, 45:65] = False
expected = skimage.restoration.denoise_bilateral(
    image=image,
    multichannel=False,
    sigma_color=sigma_range,
    sigma_spatial=sigma,
)
```
- [ ] Relevant images (if any)
- [X] Operating system and version: Ubuntu 16.04
- [X] Python version
```
>>> print(sys.version)
2.7.14 |Anaconda, Inc.| (default, Dec  7 2017, 17:05:42) 
[GCC 7.2.0]
```
- [X] scikit-image version (run `skimage.__version__`)
```
>>> skimage.__version__
'0.14.1'
```

### PR DESCRIPTION
**Pull request by jni**
## Description

Fixes #3449, again.

See https://github.com/scikit-image/scikit-image/issues/3449#issuecomment-495669637 and subsequent replies for details.

## Checklist

- [ ] ~~[Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)~~
- [ ] ~~Gallery example in `./doc/examples` (new features only)~~
- [ ] ~~Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark~~
- [x] Unit tests
- [x] Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)

## For reviewers

- [ ] Check that the PR title is short, concise, and will make sense 1 year
  later.
- [ ] Check that new functions are imported in corresponding `__init__.py`.
- [ ] Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.
- [ ] Consider backporting the PR with `@meeseeksdev backport to v0.14.x`

### CODE REVIEW

**Comment by alexandrejaguar** [comment]
Thanks @jni!

**Comment by jni** [comment]
@meeseeksdev backport to v0.14.x

**Comment by jni** [comment]
@meeseeksdev backport to v0.15.x

File: skimage/restoration/_denoise_cy.pyx [hunk 1]
```diff
@@ -119,7 +119,7 @@ def _denoise_tv_bregman(np_floats[:, :, ::1] image, np_floats weight,
         np_floats ux, uy, uprev, unew, bxx, byy, dxx, dyy, s, tx, ty
         int i = 0
         np_floats lam = 2 * weight
-        np_floats rmse = DBL_MAX
+        double rmse = DBL_MAX
```
Comment (hmaarrfk) [inline skimage/restoration/_denoise_cy.pyx]: why was this line changed, the whole point of this was to keep float32 as float32 as long as possible. wasa it causing errors?

Comment (jni) [reply skimage/restoration/_denoise_cy.pyx]: I *presume* that setting a float32 to DBL_MAX does not do good things. Perhaps I there is a FLT_MAX?

Comment (hmaarrfk) [reply skimage/restoration/_denoise_cy.pyx]: ah i see. I didn't know what DBL meant, there is a way to switch code paths between dtypes for a fused type. I guess we would need to do that here.

Comment (jni) [reply skimage/restoration/_denoise_cy.pyx]: We could just set it to FLT_MAX regardless of dtype. If your initial RMSE exceeds that, you're in deep trouble :joy:

Anyway, the code works, it's just probably a tiny bit slower because of float32-float64 comparisons in the float32 input case.

Comment (hmaarrfk) [reply skimage/restoration/_denoise_cy.pyx]: yeah, this "optimization" would be a 0.16 branch only IMO.
